### DISCUSSION TITLE
Trigger item update from custom interface

### DISCUSSION THREAD

**Discussion by iherwig** [directus/directus#11629] [Archive]
Is it possible to trigger a `PATCH` API request that saves all edits of the current item from within a custom interface? 

My particular use case is a custom interface that contains a button that triggers an external process. The external process will access item data directly from the directus database and should see the latest data of course. `this.$emit('save')` would only persist changes of the field that uses the interface, right? It would also not allow to wait for the response of the update call.

Or is there another way to accomplish this?

Thank you

**Comment by rijkvanzanten** [comment]
Hmm good question! I don't think interface extensions can currently call the form (parent level) save methods.

What `this.$emit('save');` are you currently referring too?

**Comment by iherwig** [comment]
Thanks for the reply. So i guess it's nothing you plan to support in the near future. Do you have a hint how to implement this? As far as I understand it would be necessary to access the edits on the item and then send them via a `PATCH` request to the server.

I saw `this.$emit('save')` e.g. in [field-detail-simple.vue](https://github.com/directus/directus/blob/5fc7e0ea0ecc887a15b6ecd223d11825c17fb423/app/src/modules/settings/routes/data-model/field-detail/field-detail-simple/field-detail-simple.vue) but actually do not know which code is handling the event. So it was more or less a guess that this could be related.

**Reply by rijkvanzanten** [reply]
The save action for the item detail pages is done outside of the form / interfaces editing context, so there's no real direct connection between the interfaces themselves, and the top level page that saves the staged changes. Implementing some sort of event chain that would allow interfaces themselves to trigger the parent level save operation would be non-trivial, and not something that we're currently planning on supporting.

That being said, we are currently working on a data-workflows feature that might actually solve for this use case in a slightly different manner. From what I'm understanding, you're looking to have a button that will send the current edits to the server, do some arbitrary processing, and update the form on the front-end again?

**Reply by rijkvanzanten** [reply]
Links for good measure:

The item detail save button:

https://github.com/directus/directus/blob/dea674ad323288af051c9c5d4cee692acce69cb4/app/src/modules/content/routes/item.vue#L129

The parent level composable that contains the save functionality:

https://github.com/directus/directus/blob/dea674ad323288af051c9c5d4cee692acce69cb4/app/src/modules/content/routes/item.vue#L267-L284

The actual save function:

https://github.com/directus/directus/blob/dea674ad323288af051c9c5d4cee692acce69cb4/app/src/composables/use-item/use-item.ts#L112

**Reply by iherwig** [reply]
> That being said, we are currently working on a data-workflows feature that might actually solve for this use case in a slightly different manner. From what I'm understanding, you're looking to have a button that will send the current edits to the server, do some arbitrary processing, and update the form on the front-end again?

Yes, that is exactly what I'm looking for and solving this in a more general way (like using workflows) is preferable of course. I'll be happy to test this new feature as soon as it's available. 
Thanks for the links.
